pennylane/interfaces/jax_jit.py

```diff
@@ -31,12 +31,10 @@
 
 
 def _validate_jax_version():
-    if semantic_version.match("<0.3.17", jax.__version__) or semantic_version.match(

```

Comment:
why don't you want to check minimum version anymore?

Reply:
I think it not valid anymore, my guess is that we support 0.4.1/0.4.2 and 0.4.3 but let me double check the minimal version.

Reply:
I agree with Matt. If the minimum version we support is 0.4.1 then we should also check `semantic_version.match("<0.4.1", jax.__version__)`

Reply:
So do we not support `>=0.3.17`?